internal/service/domain/tree/tree.go

```diff
@@ -62,3 +66,41 @@ func (s *TreeService) Delete(ctx context.Context, id int32) error {
 	}
 	return nil
 }
+
+func (s *TreeService) Update(ctx context.Context, id int32, tu *entities.TreeUpdate) (*entities.Tree, error) {
+	currentTree, err := s.treeRepo.GetByID(ctx, id)
+	if err != nil {
+		return nil, handleError(err)
+	}
+	// Check if the tree is readonly (imported from csv)
+	if currentTree.Readonly {
+		return nil, handleError(fmt.Errorf("tree with ID %d is readonly and cannot be updated", id))
+	}
+	fn := make([]entities.EntityFunc[entities.Tree], 0)
+	if tu.TreeClusterID != nil {
+		treeCluster, err := s.treeClusterRepo.GetByID(ctx, *tu.TreeClusterID)
+		if err != nil {
+			return nil, handleError(fmt.Errorf("failed to find TreeCluster with ID %d: %w", *tu.TreeClusterID, err))
+		}
+		fn = append(fn, tree.WithTreeCluster(treeCluster))
+	}
```

Comment:
When a tree gets updated, the old tree cluster coordinate needs to be updated as well

Reply:
see #98

Reply:
Not only the old tree cluster, but I think the new one should also be updated, right?